Assets/ComethSDK/Scripts/Core/ComethWallet.cs

```diff
@@ -52,34 +52,26 @@ public ComethWallet(IAuthAdaptor authAdaptor, string apiKey, string baseUrl = ""
 			_authAdaptor = authAdaptor;
 		}
 
-		public async Task Connect([CanBeNull] string burnerAddress = "")
+		public async Task Connect([CanBeNull] string walletAddress = "")
 		{
 			if (_authAdaptor == null) throw new Exception("No auth adaptor found");
 
 			_provider = Constants.GetNetworkByChainID(_chainId).RPCUrl;
 			_web3 = new Web3(_provider);
 
-			await _authAdaptor.Connect(burnerAddress);
+			await _authAdaptor.Connect(walletAddress);
 
 			_projectParams = await _api.GetProjectParams();
-			var account = _authAdaptor.GetAccount();
-			var predictedWalletAddress = await _api.GetWalletAddress(account);
-			_walletAddress = predictedWalletAddress ?? throw new Exception("Error while getting wallet address");
-
-			var nonce = await _api.GetNonce(predictedWalletAddress);
-			if (nonce == null) throw new Exception("Error while getting nonce");
-
-			var message = CreateMessage(predictedWalletAddress, nonce);
-			var messageToSign = SiweMessageStringBuilder.BuildMessage(message);
-			var signatureSiwe = await SignMessage(messageToSign);
-
-			//SAFE ADDRESS
-			var walletAddress = await _api.ConnectToComethWallet(
-				message,
-				signatureSiwe,
-				predictedWalletAddress
-			);
-			if (walletAddress == null) throw new Exception("Error while connecting to Cometh Wallet");
+
+			if (string.IsNullOrEmpty(walletAddress))
+			{
+				var account = _authAdaptor.GetAccount();
+				_walletAddress = await _api.GetWalletAddress(account);
```

Comment:
In both cases the walletAddress should come from the connector as it is initiated during the .connect()